geomstats/learning/frechet_mean.py

```diff
@@ -411,8 +408,11 @@ def fit(self, X, y=None, weights=None):
         self : object
             Returns self.
         """
-        is_linear_metric = isinstance(
-            self.metric, (EuclideanMetric, MatricesMetric, MinkowskiMetric))
+        metric_module = self.metric.__module__
```

Comment:
The previous version was causing a bug for metrics inheriting from e.g. `MatricesMetric` that are not actually Euclidean (e.g. Grassmann metric) because `isinstance` returns true for child classes